### PR TITLE
[core] allow default values on non-optional properties

### LINKED ISSUE #1706: Allow default values for required properties
Implementation issue for: https://github.com/microsoft/typespec/issues/1583

- [x] Compiler should not throw in this case
- [x] Update emitter docs to indicate how to use default values (should be used for documentation only)

### PR DESCRIPTION
**Pull request by tjprescott**
Fixes #1706.

### CODE REVIEW

File: docs/extending-typespec/emitters-basics.md [hunk 1]
```diff
@@ -186,3 +186,25 @@ Since an emitter is a node library, you could use standard `fs` APIs to write fi
...
+
+```ts
+const modelProp: ModelProperty = ...;   // the isActive ModelProperty type
+const defaultValue = modelProp.default; // value: true
+```
+
+It is important that emitters handle default values in a consistent way. The following guidance should be used when emitting default values:
+
+- operation parameters: use the `default` value, if provided, as the client default when no value is specified
+- model properties: do nothing. Note that some emitters may interpret default values on model properties in a way that they essentially become operation parameters. For example, a Python emitter might generate an `__init__` method that includes all model properties and their default values. In this case, the emitter should follow the guidance for operation parameters.
```
Comment (tjprescott) [inline docs/extending-typespec/emitters-basics.md:210]: @markcowl I'm not sure this is correct. The implementation issue says:
> Update emitter docs to indicate how to use default values (should be used for documentation only)

but the design issue says:
> params: use as default client values when no value is specified. properties: do nothing

Comment (timotheeguerin) [reply docs/extending-typespec/emitters-basics.md:210]: yeah that doesn't feel correct to me either, those default should never be used by the client

Comment (timotheeguerin) [reply docs/extending-typespec/emitters-basics.md:210]: or does that mean what to do if the property is required

Comment (tjprescott) [reply docs/extending-typespec/emitters-basics.md:210]: Are the current language emitters ignoring default values?

Comment (timotheeguerin) [reply docs/extending-typespec/emitters-basics.md:210]: They should be ignoring it, we have this issue to discuss client defaults https://github.com/Azure/typespec-azure/issues/800

Comment (tjprescott) [reply docs/extending-typespec/emitters-basics.md]: Okay, I updated the documentation. Please let me know if it looks right.

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**